.github/workflows/ci-checks.yml

```diff
@@ -0,0 +1,81 @@
+name: CI Checks
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+env:
+  IMAGE_NAME: "fruits-app"
+  BASE_IMAGE: "registry.access.redhat.com/ubi8/openjdk-11"
+
+
+jobs:
+  build-maven:
+    runs-on: ubuntu-latest
+    steps:
+       # Checkout Knative service deploy action github repository
+      - name: Checkout Knative service deploy action
+        uses: actions/checkout@v2
+        with:
+          path: "kn-service-deploy"
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Checkout example code
+        id: git_checkout
+        uses: actions/checkout@v2
+        with:
+          repository: kameshsampath/kn-service-action-example
+          path: 'fruits-app'
+
+      - name: Maven Build
+        id: build_app
+        working-directory: fruits-app
+        run: mvn -B -DskipTests clean package --file pom.xml
+
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - uses: actions/upload-artifact@v2
+        name: Save build target directory
+        with:
+          name: buildArtifacts
+          path: |
+            ./**/*
+
+  build-image:
+    runs-on: ubuntu-latest
+    needs: [build-maven]
+    steps:
+      - name: Download build target directory
+        uses: actions/download-artifact@v2
+        with:
+          name: buildArtifacts
+
+      - name: Commit Metadata
```

Comment:
should this be called "**Determine** or **Get** commit metadata" ? otherwise, it sounds like it's performing a `git commit`